FlowCrypt/Functionality/Services/Key Services/KeyService.swift

```diff
@@ -80,4 +81,48 @@ final class KeyService: KeyServiceType {
 
         return .success(privateKeys)
     }
+
+    func signingKey() throws -> PrvKeyInfo? {
```

Comment:
Good to have it implemented this way, in its own method. Unfortunately I cannot read the code so well - could you take a chunk of the code and write it as a private method and name it meaningfully? Also more meaningful names than `result`. And finally, we should write unit tests for this method to make sure it behaves per the logic we expected.

Otherwise looks good overall.

Reply:
Unfortunately it is not so easy to have a unit test here because `KeyService` has some dependencies like Realm objects and Core service. I did some refactoring to make code more readable.

Reply:
I think the tests will be necessary. Let's do that later in a separate PR later. There are several tests that need to be written:
 - ui test to check proper behavior when user inputs wrong pass phrase for message signing
 - ui test to check proper behavior when user eventually inputs correct pass phrase
 - ui test to check proper behavior when pass phrase already in memory
 - unit tests to make sure that the right private key is chosen in various scenarios